### PR TITLE
feature: Add support for OpenCover report format

### PR DESCRIPTION
**Pull request by franciscodua**

### CODE REVIEW

File: src/main/scala/com/codacy/parsers/implementation/OpenCoverParser.scala [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    val validMethods = (rootNode \\ "Method").filter(m => (m \ FileRefTag).nonEmpty)
+
+    val fileReports = (for {
+      (fileIndex, methods) <- validMethods.groupBy(m => (m \ FileRefTag \@ IdAttribute).toInt)
+      filename <- fileIndices.get(fileIndex)
+
+      sanitisedFileName = TextUtils.sanitiseFilename(filename).stripPrefix(projectRoot).stripPrefix("/")
+      lineCoverage = getLineCoverage(methods, sanitisedFileName)
+      totalLines = lineCoverage.size
+      coveredLines = lineCoverage.count(_._2 > 0)
```
Comment (mrfyda) [inline src/main/scala/com/codacy/parsers/implementation/OpenCoverParser.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/gh/codacy/coverage-parser/pullRequest?prid=4948206)

File: src/main/scala/com/codacy/parsers/implementation/OpenCoverParser.scala [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    loadXml(reportFile).right.flatMap { r =>
+      Try(parseReportNode(r, TextUtils.sanitiseFilename(rootProject.getAbsolutePath))) match {
+        case Success(coverageReport) => Right(coverageReport)
+        case Failure(ex) => Left(s"Failed to parse report with error: ${ex.getMessage}")
+      }
+    }
+  }
+
+  private def parseReportNode(rootNode: NodeSeq, projectRoot: String): CoverageReport = {
+    val fileIndices: Map[Int, String] = (rootNode \\ "Files" \ "File").map { n =>
```
[the remark this answers is not in the document]
Comment (franciscodua) [reply src/main/scala/com/codacy/parsers/implementation/OpenCoverParser.scala]: Done ✅

File: src/main/scala/com/codacy/parsers/implementation/OpenCoverParser.scala [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    val (totalLines, coveredLines) = fileReports
+      .foldLeft((0, 0)) {
+        case ((total, covered), f) =>
+          val totalLines = f.coverage.size
+          val coveredLines = (f.total * totalLines) / 100
+          (total + totalLines, covered + coveredLines)
+      }
+
+    val totalCoverage = computePercentage(coveredLines, totalLines)
+    totalCoverage
```
[the remark this answers is not in the document]
Comment (franciscodua) [reply src/main/scala/com/codacy/parsers/implementation/OpenCoverParser.scala]: Done

File: src/main/scala/com/codacy/parsers/implementation/OpenCoverParser.scala [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      filename <- fileIndices.get(fileIndex)
+
+      sanitisedFileName = TextUtils.sanitiseFilename(filename).stripPrefix(projectRoot).stripPrefix("/")
+      lineCoverage = getLineCoverage(methods, sanitisedFileName)
+      totalLines = lineCoverage.size
+      coveredLines = lineCoverage.count { case (_, visitCount) => visitCount > 0 }
+      coverage = computePercentage(coveredLines, totalLines)
+    } yield {
+      CoverageFileReport(sanitisedFileName, coverage, lineCoverage)
+    }).toSeq
```
Comment (lolgab) [inline src/main/scala/com/codacy/parsers/implementation/OpenCoverParser.scala]: The yield doesn't need braces

File: src/main/scala/com/codacy/parsers/implementation/OpenCoverParser.scala [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    CoverageReport(totalCoverage, fileReports)
+  }
+
+  private def getLineCoverage(methodNodes: NodeSeq, filename: String) = {
+    val lineCoverage = for {
+      methodNode <- methodNodes
+      sequencePoint <- methodNode \\ SequencePointTag
+    } yield {
+      (sequencePoint \@ LineAttribute).toInt -> (sequencePoint \@ VisitCounterAttribute).toInt
+    }
```
Comment (lolgab) [inline src/main/scala/com/codacy/parsers/implementation/OpenCoverParser.scala]: Same here. no need of braces

**Comment by lolgab** [comment]
Before merging remove those braces

File: src/main/scala/com/codacy/parsers/implementation/OpenCoverParser.scala (hunk 5 above)
Comment (franciscodua) [reply src/main/scala/com/codacy/parsers/implementation/OpenCoverParser.scala]: Done

File: src/main/scala/com/codacy/parsers/implementation/OpenCoverParser.scala (hunk 4 above)
Comment (franciscodua) [reply src/main/scala/com/codacy/parsers/implementation/OpenCoverParser.scala]: Done

**Pull request merged**